Sqlancer report error: Getting analyzing error from line 1, column 591 to line 1, column 632. Detail message: Column type ARRAY<varchar(65533)> does not support binary predicate operation.

### Steps to reproduce the behavior (Required)
1. Prepare Data:

```sql
-- Time: 2023/04/23 17:12:05
-- Database: p14
-- Database version: 5.1.0
-- seed value: 1682240384399
DROP DATABASE IF EXISTS p14;
CREATE DATABASE p14;
USE p14;
admin set frontend config ("enable_decimal_v3" = "true");
CREATE TABLE IF NOT EXISTS t0 (c_0_0 BOOLEAN NOT NULL,c_0_1 DATE NOT NULL,c_0_2 DECIMAL(20, 4)  NOT NULL,c_0_3 VARCHAR(21) NOT NULL,c_0_4 ARRAY<STRING> NOT NULL,c_0_5 DECIMAL(18, 17)  NOT NULL,c_0_6 VARCHAR(21) NOT NULL,c_0_7 DATE NOT NULL,c_0_8 VARCHAR(41) NOT NULL,c_0_9 DECIMAL(7, 0)  NOT NULL,c_0_10 DOUBLE NOT NULL,c_0_11 BOOLEAN NOT NULL,c_0_12 BOOLEAN NOT NULL,c_0_13 DOUBLE NOT NULL,c_0_14 TINYINT NOT NULL,c_0_15 VARCHAR(1) NOT NULL,c_0_16 ARRAY<INT> NOT NULL,c_0_17 ARRAY<STRING> NOT NULL, INDEX index1_c_0_8(`c_0_8`) USING BITMAP) PRIMARY KEY (c_0_0,c_0_1) PARTITION BY RANGE(c_0_1) (START ("2010-01-01") END ("2022-12-31") EVERY (INTERVAL 197 day)) DISTRIBUTED BY HASH (c_0_0,c_0_1) BUCKETS 3 properties("replication_num"="1") ;
CREATE TABLE IF NOT EXISTS t1 (c_1_0 VARCHAR(11) NOT NULL,c_1_1 TINYINT NOT NULL,c_1_2 DATE NOT NULL,c_1_3 VARCHAR(11) NOT NULL,c_1_4 BOOLEAN NOT NULL,c_1_5 DATE NOT NULL,c_1_6 ARRAY<INT> NOT NULL,c_1_7 ARRAY<STRING> NOT NULL,c_1_8 DECIMAL(8, 1)  NOT NULL,c_1_9 DOUBLE NOT NULL,c_1_10 DECIMAL(26, 10)  NOT NULL,c_1_11 ARRAY<INT> NOT NULL) PRIMARY KEY (c_1_0,c_1_1,c_1_2,c_1_3,c_1_4,c_1_5) PARTITION BY RANGE(c_1_2) (START ("2010-01-01") END ("2022-12-31") EVERY (INTERVAL 593 day)) DISTRIBUTED BY HASH (c_1_3,c_1_1,c_1_2,c_1_0,c_1_4) BUCKETS 3 properties("replication_num"="1") ;
CREATE TABLE t2 (c_2_0 DATETIME NOT NULL ,c_2_1 DATETIME NULL ,c_2_2 VARCHAR(31) NOT NULL ,c_2_3 DATETIME NULL ,c_2_4 DATE NOT NULL ,c_2_5 VARCHAR(41) NULL ,c_2_6 BOOLEAN NOT NULL ,c_2_7 BOOLEAN NULL ,c_2_8 INT NULL ,c_2_9 VARCHAR(21) NOT NULL ,c_2_10 ARRAY<STRING> NOT NULL ,c_2_11 LARGEINT NOT NULL ) DUPLICATE KEY (c_2_0,c_2_1,c_2_2,c_2_3,c_2_4,c_2_5,c_2_6,c_2_7,c_2_8) PARTITION BY RANGE(c_2_0) (START ("2010-01-01") END ("2022-12-31") EVERY (INTERVAL 316 day)) DISTRIBUTED BY HASH (c_2_2,c_2_6,c_2_0) BUCKETS 3 properties("replication_num"="1", "bloom_filter_columns"="c_2_9") ;
INSERT INTO t0 VALUES (false, '2010-01-11', 101156010113.1019, 'Eq', ["l", "", ""], 3E-17, 'Mr7dC', '2010-01-02', '', 3291194, 0.43633804273819654, false, true, 0.43633804273819654, 0, '', [810360286, -1749242159, 810360286, 693221355, 693221355, 1722100716, 643765781, -789135626, -1393166946, 1680606165, 2113583456, -1588435069, 259231942, -55226116, -1588435069], ["", "QcGcuZMul", "3", "W", "vC", "kgnn", "3dg", "", "", "g", "", "2", "aH5vt", "LAi6Uf", "j"]), (true, '2010-01-04', 107376857400.9266, '', [], 8E-17, 'xcUGhdqqx', '2010-01-23', 'P', 1327451, 0.07885482327370352, true, true, 0.43633804273819654, 5, '', [1582470905, -876827136, -505724622, 1307223838, 643765781, -872145930, -1582695866, 1436230231], ["", "1F", "C", "ZgeUsdM", "MBikW9Uuh", "0hT37BZY", "9Hg", "p8vy", "K6F5Kwm", "8bjPKt", "ud", ""]), (false, '2010-01-04', 670042479942.3401, '', ["Rld82JzW", "w8", "is9y", "tC", "", "PJ6", "j", "dpD", "UxlxDBNm", "k", "b6"], 4E-17, 'P', '2010-01-22', 'b', 1035476, 0.6905150172152117, true, false, 1.722100716E9, 6, '', [-9223372036854775808, -870284094, 1436230231, 1307223838, -876827136, 1688934224, 2116379309, -288047770, 1307223838, 496754462, 1688934224, 1600718346, 1307223838, -876827136], ["", "E", "", "w5s", "TlH", "", "", "oXSnLcm", "", "s", "POYfGf", "1"]);
INSERT INTO t2 VALUES ('2010-01-08 08:00:14', '2010-01-29 08:00:14', 'b0', '2009-12-31 08:00:14', '2010-01-18', 'KE', true, false, 24, '', ["", "ENLO", ""], -727780094);
INSERT INTO t0 VALUES (true, '2010-01-10', 772808443239.1516, 'h', ["1vCZzju5V"], 3E-17, 'sIxB', '2010-01-04', 'G6h5', 5151401, 2.113583456E9, true, true, 0.07885482327370352, 0, '', [173329598, -1753619720, -896723146], ["m", "", ""]), (true, '2010-01-15', 539726103474.6534, 'UA', ["", "", "l", "MIR", "Np", "", "", "z", "XlcI03R", "wGaZ", "", "zU", "s", "eRAUvh6"], 5E-17, '', '2010-01-20', '', 3889521, 0.07885482327370352, false, false, -8.96723146E8, 1, '', [], []);
INSERT INTO t1 VALUES ('V', 2, '2010-01-14', '', true, '2010-01-02', [460274876, -1582695866, 1744294458, 895340951, -170340466, -1060079097, 994660606, -980170803, 496754462, -735053828, -149850287], ["C", "", "uu", "", "vGRjaFmY", "", "vi", "flHh", "", "YyQb2ut", "m", "", "M", "d0pyn", ""], 353627.1, 0.6905150172152117, 310148.6329148666, [-922371725, -1212845834, 994660606, -1896917597, 1156099606, 259231942, 1713364651, -922371725, 259231942, 1257389065, -25037575, 1250913999, -2122367056, -9223372036854775808]), ('m4ax', 2, '2010-01-09', '', false, '2010-01-25', [-1040973546, 548165941, 1713364651, -1724123797, -1035408592, 1780569520, -872145930, -1955253012, 2046992249], ["aFHo1ua1", "T1", "YdN", "kHAV33", "9HNx9eJ", "MRPO", "0", "q", "oxQ", "gpmA", "x", "JwAKHET", "B9OF3iTv", "wO", ""], 588920.3, 0.06887550411584586, 588436.4281561389, [1380993014, 1554373774, 1680918794, 235127457, -363414290, 201130996, 1066693219, 1680918794, -1896917597, -1071331331, -715285028]), ('XQJu', 2, '2010-01-05', 'AKrw', false, '2010-01-17', [1478088852, 446594393, -1807671603, -93290115, 1307223838, -1582695866, -1657337746, -1660124743, -1071331331, 1453267538], [""], 125274.0, 0.6775869669898508, 38092.1230486843, [1539672457]);
INSERT INTO t1 VALUES ('mlGFjAm', 1, '2010-01-26', '', true, '2010-01-10', [1442708588, 1156099606, 681804143, 801818043, 1380993014], [], 588487.3, -9.3290115E7, 160742.0918960880, [1936127517, 810360286, -1228144992, -313252261, -1273474429]), ('Xh7w', 2, '2010-01-07', 'o', true, '2010-01-02', [-1969753007, -727780094, -1896917597, 1, -1376363426], ["Q9LPDi", "t", "M", "ci", "", "9f", "Y8HsO", "7W", "yP", "Xo"], 191523.5, 0.08417821320594654, 957774.1476382605, [409386035, -1376363426, 2116379309, -544680053, -1331373913, -1054967851, 761902697, 810360286, 1232089864, 1380993014]);
TRUNCATE TABLE t2;
INSERT INTO t2 VALUES ('2010-01-19 08:00:14', '2010-01-12 08:00:14', 'u', '2010-01-31 08:00:14', '2010-01-11', '', true, false, 16, 'bNpgKBA', ["", "", "Jm", "", "TxKB"], 1467834380), ('2010-01-01 08:00:14', '2009-12-31 08:00:14', 'VfPXnebc', '2010-01-12 08:00:14', '2010-01-27', 'WLj67JI', false, false, 15, 'wZ8ViO6MF', ["J3C", "", "F5", "Z0BtEAHCC", "jU5T", "T", "y"], 1257389065);
INSERT INTO t1 VALUES ('', 4, '2010-01-27', 'U', true, '2010-01-27', [1611480684, 1379837123, 681951714, -91736740, -1896917597, -162923399, 77543840, 749238314, 852316171, 1251100725], ["h2", "5", ""], 283845.5, 0.08417821320594654, 605098.9150572637, [680325006, 1500060233]);
INSERT INTO t0 VALUES (false, '2010-01-07', 886786909044.4860, 'cvF0', ["HYj"], 4E-17, '', '2010-01-14', '7', 2695510, 0.4459803041364999, true, true, 0.39582672451341394, 0, '', [9223372036854775807, 1436230231, 53263714, 1980277104, -130500157], ["Sh4E9", "HGs6", "U3YfQq", "", "Dm", "N4", "pp", "", "", "AnO87m2LH", "v8PTgp6O0", "Yj", "D0OKxlt1F", "4b"]), (true, '2010-01-23', 305681386945.7937, 'oA', ["", "yOO1", "", "gd", "", "K", "bk4dza0A", "8N8Li", "yQ0", "MbV0AECs", "OHr", "dwtGu", "dT"], 6E-17, 'Wm3F4', '2010-01-02', '', 8236105, 7.49238314E8, false, true, 0.0508248177342846, 6, '', [852911231, -367779822, 985027860, 1106859449, 1500060233], ["", "Pwxmm0Jv", "hF", "gQy0dH8gQ", "0i9UjH", "h", "Jy", ""]), (false, '2010-01-04', 82170081588.7705, '', ["q", "P0q", "JIk2HgVH", "gp", "B", "XS0", ""], 6E-17, '', '2010-01-24', '', 7407484, -7.27780094E8, false, false, 0.19129465846216087, 6, '', [1467834380, 1089953213, -1543240967, -313252261, 807052039, -1615355320, -68292294, 1495381671, 53263714, -869258512], ["", "t", "Y", "mDB", "UrU", "", "WFmalw8u", "ho"]);
INSERT INTO t0 VALUES (false, '2010-01-06', 548878811749.0075, 'jr', ["13", "l", "", "G", "W", "6", "0uVqqJ", "iYHxDsJW", "gRGD6H4", "G", "pa20", "ZCGk5B3K", ""], 5E-17, '', '2010-01-19', 'xw23', 8286535, 0.7923834251376212, false, false, 0.1938022785759953, 7, '', [201383536, -1724123797], ["cmvL", "d", "mEK", "3C0", "elf8l94", "8T", "RD7g5nR"]), (false, '2010-01-30', 122486091428.5032, 'w', ["d0"], 7E-17, '', '2010-01-27', '94eKo', 9706670, 0.10139572393616525, false, true, 0.37251942113150516, 6, '', [224580135, -363414290], ["", "", "6cPyT", "X3JcuupE", "ZOG", "", "k9HAq", "gmtqs"]);
INSERT INTO t0 VALUES (false, '2010-01-23', 832171867839.9291, 'vd', ["aApR", "fb", "fDA", "sDOAiz", "Ab0", "JDP6In66", "", "", "Blz5jUaic", "QuuYq"], 2E-17, 'rl', '2010-01-25', '36b9CkcJb', 5642213, 0.6081428162171885, true, false, -1.228144992E9, 4, '', [801818043, -1275113330, -1035408592, 299852126, 293932989, -1646632574, -1313185248, -1001171712], ["CXk8f", "lG", "", ""]), (false, '2010-01-24', 46925159496.3873, '', ["qrP", "dy5Q760", "xiQ", "j", "JcOB", "iF", "KP6", "", "n"], 5E-17, '0E9hoE', '2009-12-31', 'HD0wD1Wo', 7976639, 0.09118967904197217, true, true, 0.6945722518256796, 4, '', [1241397941, -884517302, 71294729, 1637786248, 1152680174, 1611480684, 247484268, -51090602, -781958673], ["Ew3uqx", "66QZG", "qcAN", "NVEJlK", "x", "", "H", "iU", "Hf", "", "Uj", "", "", "V"]);
INSERT INTO t2 VALUES ('2010-01-18 08:00:14', '2010-01-04 08:00:14', 'BKVoLN', '2010-01-25 08:00:14', '2010-01-28', 'L', false, false, 24, 'Pn8EB4', ["", "", "H0bH", "", "t", "IjWzl"], 693221355), ('2010-01-02 08:00:14', '2010-01-18 08:00:14', 'vc', NULL, '2010-01-22', 'sYlO2o', false, true, 31, 'o', ["43NHv4C0", "1", "KiToE", "", "jTBTGXlF", "KeVbXbQd", "ab9x5", "Z", "eA", "dS"], 852316171), ('2010-01-20 08:00:14', '2010-01-04 08:00:14', 'N', '2010-01-19 08:00:14', '2010-01-13', 'rV', true, NULL, 17, '', ["E", "3A6", "s", "", "", "wo4q", "c", "Eme", "", "MlKsOqN", "i", "Iw", "ms"], -46887), ('2010-01-05 08:00:14', '2010-01-30 08:00:14', 'A1tVF', '2010-01-30 08:00:14', '2010-01-02', NULL, false, false, 19, '', ["zJ", "", "", "Vu", "", "", ""], 1550047956);
INSERT INTO t1 VALUES ('Ie', 5, '2010-01-31', 'OzYB', false, '2010-01-03', [1342856288], ["J7", "JfdbC", "sc", "", "k", "", "SBX", "X", "MUdt", "FePNG", "ri2jl", "5L3epTt4", "x2S7", "V4", "zSKquU"], 624735.6, 0.7969648701604166, 559837.2240876159, [-1166936816, 901306381, 1330325369, 749854864, 1758129870, -1749242159, -838641411, -1209893689, 1338012349]), ('', 0, '2010-01-06', 'QX', false, '2009-12-31', [-1391571295, -487745989, 911299751, 718326690, 871286357, 1426062538, 1112521372, -1660124743, -2014099552, 1554373774, 1732203785, -1075247145, -735053828], ["l", "RJ", "", "N", "Cy4M", "IkcyAIR1Y"], 51777.5, 4.46594393E8, 396469.5761843678, [-505724622, 1166037897, -701530615, -291220426, 1858209343, 173329598, 334713034, 496754462, -755100488, -15354336, 446594393, 1680918794, -1611492464, -170340466, 59075499]);
INSERT INTO t1 VALUES ('eBJ', 4, '2010-01-30', 'eyY', false, '2010-01-08', [1902398881, -715285028, 1937298765, -661444531, 681951714, 1965706782, -1, 1244813693, -1753619720, -511567170, -2144166859], [], 869266.9, 0.6905150172152117, 237679.9823394489, [-42957206]), ('Hs', 7, '2010-01-26', 'Y2', true, '2010-01-20', [483768486, -955739215, -1179419470, 434718166, 884355283, -800753880, -1896917597, 2116379309, 1043200782, 1392606829, 136782136], ["4mmE7I", "jqHJ6i", "fR75rZy", "", "", "mZA"], 230172.3, -5.5226116E7, 560600.7245803829, [-313614108]);
INSERT INTO t0 VALUES (true, '2010-01-26', 665361556575.6073, 'oTEWo', ["gAHIJhv", "L", "7VMC0IJ", "k", "kHMavPide", "incE", "T", "L", "q", "SGe"], 1E-17, '', '2010-01-15', 'khsD', 9779872, 0.3907237203548056, false, true, 0.8335091835679848, 4, '', [-922371725, 2144057029, -1228144992, 423734678, 643765781, -976125685, 1913144740, 311204932, -853150352, -1102680167, 1722100716], ["1", "p1AX0", "PUu5", "PO", "XL70mANZW", "dN4802xCk", "", "0WwN6ETUp", "p5", "hxbvisCA", "tpITDOuKE", "BHx", "ml", "VL3", "Dz"]);
INSERT INTO t2 VALUES ('2010-01-19 08:00:14', NULL, 'Ldq5T', '2010-01-11 08:00:14', '2010-01-04', '8RVwfD', true, false, 6, 'vd0', ["EgnQ", "W", "", "", "", "kBIDpSVG", "", "", "pS", "9Lscho", "", "1", "Dbw", "", "8"], 259231942), ('2010-01-31 08:00:14', '2010-01-24 08:00:14', 'U', '2010-01-03 08:00:14', '2010-01-31', 'M', false, false, 28, 'xp', ["", "m", "v", "QNbZ", "cAWDC", "", "5N", "tI", "b"], 896560934), ('2010-01-16 08:00:14', '2010-01-11 08:00:14', '', NULL, '2010-01-21', 'gA7QA', false, false, 19, '', ["tlqptw", "cs"], -55226116);
INSERT INTO t2 VALUES ('2010-01-12 08:00:14', '2010-01-23 08:00:14', '', '2010-01-09 08:00:14', '2010-01-14', 'iu', false, true, 22, '8CQ1HvN', ["", "wOOyUfrfy", "pBha"], 810360286), ('2010-01-24 08:00:14', '2010-01-16 08:00:14', 'v2HKfj7x', '2009-12-31 08:00:14', '2010-01-10', 'BTR', false, false, 14, '4', ["", "Egll", "d", ""], 1730766932);
INSERT INTO t0 VALUES (false, '2010-01-24', 144423649384.9032, '1H', ["qx", "", "D9Gn", "O32qA", "J", "", "WiA6FX", "e2z", "A", "", "", "HSm46jPB"], 2E-17, 'i', '2010-01-19', '', 490895, 0.5830277856588745, true, true, 0.3451548876029824, 7, '', [1127931083, -1071331331], ["", "Gh", "", "Jcz", "0", "v", "FOUJ1uN0v", "tmDwP0A3L"]);
INSERT INTO t0 VALUES (false, '2010-01-11', 285928684745.9536, 'Yle', ["MnG"], 6E-17, '', '2010-01-17', 'kQffqgMv', 111325, 0.8193582950143077, true, false, 0.6378003492685722, 6, '', [643765781, 1833777322, 1992308711, 752347846, -1353110449, -711137407, -149850287, -1264551587], ["HWA40c", "X", "", "krHN", "WPb", "VU25", "9", "yp", "BjmVUn", "", "ENT9hUN", "r", "", "gDr", "dK"]), (false, '2010-01-20', 763502593350.7858, '7', ["Z", "", "fYou", "4tGtIw", "MG", "ZLcTgfl", "Se6RYfan6", "WO9Sn7", "eMMqnON", "", "2Vc", "m6jZ"], 3E-17, 'S', '2010-01-30', 'SA', 3499848, 0.3433834707809641, true, false, 0.8978831881253633, 3, '', [1453267538, -1899420326, 1257389065, 1156164681, 1780708085, -1588435069, -1071331331, 1912208211, -148753127, 693221355, -2106429530, 761298099, -170340466, -1393166946], ["", "See", "5", "IlQ3cV", "", "Cs84j", "", "", "SI"]);
INSERT INTO t2 VALUES ('2010-01-12 08:00:14', '2010-01-09 08:00:14', '', '2010-01-11 08:00:14', '2010-01-29', NULL, true, false, 2, '15', ["C8Sr9bY", "X", "ltEIt", "zvf"], 1339461543);
SELECT * FROM (SELECT DISTINCT t2_271.c_2_0, t2_271.c_2_1, t2_271.c_2_2, t2_271.c_2_3, t2_271.c_2_4, t2_271.c_2_5, t2_271.c_2_6, t2_271.c_2_7, t2_271.c_2_8, t2_271.c_2_9, t2_271.c_2_10, t2_271.c_2_11 FROM t2 AS t2_271 WHERE true ORDER BY t2_271.c_2_7 ASC, t2_271.c_2_6 DESC) as result WHERE result.c_2_0 = '2010-01-19 08:00:14' AND result.c_2_1 = '2010-01-12 08:00:14' AND result.c_2_2 = 'u' AND result.c_2_3 = '2010-01-31 08:00:14' AND result.c_2_4 = '2010-01-11' AND result.c_2_5 = '' AND result.c_2_6 = true AND result.c_2_7 = false AND result.c_2_8 = 16 AND result.c_2_9 = 'bNpgKBA' AND result.c_2_10 = ["", "", "Jm", "", "TxKB"] AND result.c_2_11 = 1467834380;

```

2. Query

```sql
SELECT * FROM (SELECT DISTINCT t2_271.c_2_0, t2_271.c_2_1, t2_271.c_2_2, t2_271.c_2_3, t2_271.c_2_4, t2_271.c_2_5, t2_271.c_2_6, t2_271.c_2_7, t2_271.c_2_8, t2_271.c_2_9, t2_271.c_2_10, t2_271.c_2_11 FROM t2 AS t2_271 WHERE true ORDER BY t2_271.c_2_7 ASC, t2_271.c_2_6 DESC) as result WHERE result.c_2_0 = '2010-01-19 08:00:14' AND result.c_2_1 = '2010-01-12 08:00:14' AND result.c_2_2 = 'u' AND result.c_2_3 = '2010-01-31 08:00:14' AND result.c_2_4 = '2010-01-11' AND result.c_2_5 = '' AND result.c_2_6 = true AND result.c_2_7 = false AND result.c_2_8 = 16 AND result.c_2_9 = 'bNpgKBA' AND result.c_2_10 = ["", "", "Jm", "", "TxKB"] AND result.c_2_11 = 1467834380;
```
### Expected behavior (Required)
NO Error

### Real behavior (Required)
Error as follows
```
--java.lang.AssertionError: SELECT * FROM (SELECT DISTINCT t2_271.c_2_0, t2_271.c_2_1, t2_271.c_2_2, t2_271.c_2_3, t2_271.c_2_4, t2_271.c_2_5, t2_271.c_2_6, t2_271.c_2_7, t2_271.c_2_8, t2_271.c_2_9, t2_271.c_2_10, t2_271.c_2_11 FROM t2 AS t2_271 WHERE true ORDER BY t2_271.c_2_7 ASC, t2_271.c_2_6 DESC) as result WHERE result.c_2_0 = '2010-01-19 08:00:14' AND result.c_2_1 = '2010-01-12 08:00:14' AND result.c_2_2 = 'u' AND result.c_2_3 = '2010-01-31 08:00:14' AND result.c_2_4 = '2010-01-11' AND result.c_2_5 = '' AND result.c_2_6 = true AND result.c_2_7 = false AND result.c_2_8 = 16 AND result.c_2_9 = 'bNpgKBA' AND result.c_2_10 = ["", "", "Jm", "", "TxKB"] AND result.c_2_11 = 1467834380;
--	at sqlancer.common.query.QueryAdapter.checkException(QueryAdapter.java:98)
--	at sqlancer.common.query.QueryAdapter.executeAndGet(QueryAdapter.java:128)
--	at sqlancer.common.oracle.PivotedQuerySynthesisBase.containsRows(PivotedQuerySynthesisBase.java:66)
--	at sqlancer.common.oracle.PivotedQuerySynthesisBase.check(PivotedQuerySynthesisBase.java:49)
--	at sqlancer.ProviderAdapter.generateAndTestDatabase(ProviderAdapter.java:55)
--	at sqlancer.Main$DBMSExecutor.run(Main.java:334)
--	at sqlancer.Main$2.run(Main.java:542)
--	at sqlancer.Main$2.runThread(Main.java:519)
--	at sqlancer.Main$2.run(Main.java:509)
--	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
--	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
--	at java.lang.Thread.run(Thread.java:748)
--Caused by: com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: Getting analyzing error from line 1, column 591 to line 1, column 632. Detail message: Column type ARRAY<varchar(65533)> does not support binary predicate operation.
--	at sun.reflect.GeneratedConstructorAccessor6.newInstance(Unknown Source)
--	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
--	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
--	at com.mysql.jdbc.Util.handleNewInstance(Util.java:425)
--	at com.mysql.jdbc.Util.getInstance(Util.java:408)
--	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:943)
--	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3973)
--	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3909)
--	at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2527)
--	at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2680)
--	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2497)
--	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2455)
--	at com.mysql.jdbc.StatementImpl.executeQuery(StatementImpl.java:1369)
--	at sqlancer.common.query.QueryAdapter.executeAndGet(QueryAdapter.java:118)
--	... 10 more
--
```
### StarRocks version (Required)
- main(688433861d)

--- a/fe/fe-core/src/main/java/com/starrocks/analysis/BinaryPredicate.java
+++ b/fe/fe-core/src/main/java/com/starrocks/analysis/BinaryPredicate.java
@@ -294,7 +294,7 @@ public class BinaryPredicate extends Predicate implements Writable {
         }
         if (type1.isArrayType() || type2.isArrayType()) {
             // Must both be array
-            if (!type1.equals(type2)) {
+            if (!type1.matchesType(type2)) {
                 return Type.INVALID;
             }
             return type1;

--- a/test/sql/test_array/R/test_array
+++ b/test/sql/test_array/R/test_array
@@ -77,3 +77,55 @@ select c4 > c5 from array_data_type;
 0
 1
 -- !result
+-- name: testArrayVarchar
+CREATE TABLE array_data_type_1
+    (c1 int,
+    c2  array<datetime>,
+    c3  array<float>,
+    c4  array<varchar(10)>,
+    c5  array<varchar(20)>,
+    c6  array<array<varchar(10)>>)
+    PRIMARY KEY(c1)
+    DISTRIBUTED BY HASH(c1)
+    buckets 1
+    PROPERTIES ("replication_num" = "1");
+-- result:
+-- !result
+insert into array_data_type_1 values
+(1, ['2020-11-11', '2021-11-11', '2022-01-01'], [1.23, 1.35, 2.7894], ['a', 'b'], ['sss', 'eee', 'fff'], [['a', 'b']]),
+(2, ['2020-01-11', null, '2022-11-01'], [2.23, 2.35, 3.7894], ['aa', null], ['ssss', 'eeee', null], [['a', null], null]),
+(3, null, null, null, null, null);
+-- result:
+-- !result
+select * from array_data_type_1 where c4 != ['a'] or c6 = [['a', 'b']];
+-- result:
+1	["2020-11-11 00:00:00","2021-11-11 00:00:00","2022-01-01 00:00:00"]	[1.23,1.35,2.7894]	["a","b"]	["sss","eee","fff"]	[["a","b"]]
+2	["2020-01-11 00:00:00",null,"2022-11-01 00:00:00"]	[2.23,2.35,3.7894]	["aa",null]	["ssss","eeee",null]	[["a",null],null]
+-- !result
+select * from array_data_type_1 where c4 = ['a'] or c6 != [['a', 'b']];
+-- result:
+2	["2020-01-11 00:00:00",null,"2022-11-01 00:00:00"]	[2.23,2.35,3.7894]	["aa",null]	["ssss","eeee",null]	[["a",null],null]
+-- !result
+select * from array_data_type_1 where c4 = cast(c4 as array<char(10)>);
+-- result:
+1	["2020-11-11 00:00:00","2021-11-11 00:00:00","2022-01-01 00:00:00"]	[1.23,1.35,2.7894]	["a","b"]	["sss","eee","fff"]	[["a","b"]]
+2	["2020-01-11 00:00:00",null,"2022-11-01 00:00:00"]	[2.23,2.35,3.7894]	["aa",null]	["ssss","eeee",null]	[["a",null],null]
+-- !result
+select * from array_data_type_1 where c5 = c4 or c6 = [['a']];
+-- result:
+-- !result
+select * from array_data_type_1 where array_map((x) -> concat(x, 'a'), c5) = c4;
+-- result:
+-- !result
+select c6[0] = ['a'] from array_data_type_1;
+-- result:
+None
+None
+None
+-- !result
+select c6[0] > array_map((x) -> concat(x, 'a'), c5) from array_data_type_1;
+-- result:
+None
+None
+None
+-- !result

--- a/test/sql/test_array/T/test_array
+++ b/test/sql/test_array/T/test_array
@@ -33,3 +33,29 @@ insert into array_data_type (c1, c2, c3, c4,c5) values
 
 select c4 = c5 from array_data_type;
 select c4 > c5 from array_data_type;
+
+-- name: testArrayVarchar
+CREATE TABLE array_data_type_1
+    (c1 int,
+    c2  array<datetime>,
+    c3  array<float>,
+    c4  array<varchar(10)>,
+    c5  array<varchar(20)>,
+    c6  array<array<varchar(10)>>)
+    PRIMARY KEY(c1)
+    DISTRIBUTED BY HASH(c1)
+    buckets 1
+    PROPERTIES ("replication_num" = "1");
+
+insert into array_data_type_1 values
+(1, ['2020-11-11', '2021-11-11', '2022-01-01'], [1.23, 1.35, 2.7894], ['a', 'b'], ['sss', 'eee', 'fff'], [['a', 'b']]),
+(2, ['2020-01-11', null, '2022-11-01'], [2.23, 2.35, 3.7894], ['aa', null], ['ssss', 'eeee', null], [['a', null], null]),
+(3, null, null, null, null, null);
+
+select * from array_data_type_1 where c4 != ['a'] or c6 = [['a', 'b']];
+select * from array_data_type_1 where c4 = ['a'] or c6 != [['a', 'b']];
+select * from array_data_type_1 where c4 = cast(c4 as array<char(10)>);
+select * from array_data_type_1 where c5 = c4 or c6 = [['a']];
+select * from array_data_type_1 where array_map((x) -> concat(x, 'a'), c5) = c4;
+select c6[0] = ['a'] from array_data_type_1;
+select c6[0] > array_map((x) -> concat(x, 'a'), c5) from array_data_type_1;

